tests/util/dm_tests.rs

```diff
@@ -72,10 +78,12 @@ pub fn test_thinpool_device(paths: &[&Path]) -> () {
                                               initialized.last().unwrap());
 
     let dm = DM::new().unwrap();
-    let metadata_dev = LinearDev::new("stratis_testing_thinpool_metadata",
-                                      &dm,
-                                      vec![metadata_blockdev.avail_range()])
-            .unwrap();
+    let (meta_start, meta_length) = metadata_blockdev.borrow().avail_range();
```

Comment:
I think that, in theory, all the newly introduced borrowing could be dropped in this file.